src/js/angular/sparql-editor/controllers.js

```diff
@@ -79,11 +85,29 @@ function SparqlEditorCtrl($scope,
                         const canceler = $q.defer();
                         return autocompleteLocalNames(term, canceler);
                     }
-                }
+                },
+                getRepositoryStatementsCount: getRepositoryStatementsCount
             };
         }
     };
 
+    const getRepositoryStatementsCount = () => {
+        return new Promise(function (resolve) {
```

Comment:
This method has a strange construction. Why do you need to wrap the promise returned by `RDF4JRepositoriesRestService.getRepositorySize()` into another promise?

Reply:
`RDF4JRepositoriesRestService.getRepositorySize()` returns `HttpPromise` which is a bit different than `Promise`. We can not  chain `HttpPromise` as we expected. Here is an article that describes the problems [AngularJS HttpPromise methods break promise chain](https://medium.com/@ExplosionPills/angularjs-httppromise-methods-break-promise-chain-950c85fa1fe7)
Other way to write the function is:
`const getRepositoryStatementsCount = () => {
        return RDF4JRepositoriesRestService.getRepositorySize()
            .then(function (response) {
                const number = parseInt(response.data);
                return $q.when(number);
            })
            .catch(function (data) {
                const params = {
                    repo: $repositories.getActiveRepository(),
                    error: getError(data)
                };
                toastr.warning($translate.instant('query.editor.repo.size.error', params));
            });
    };`